### ISSUE TITLE
Hardcoded paths

### ISSUE THREAD

**Issue by astraszab** [StanislavYatsevich/Store_Sales#2]
https://github.com/StanislavYatsevich/Store_Sales/blob/29e25693b941653d592edbd11e2a45ab4d626282/src/store_sales/streamlit_app.py#L7

Some of the possible solutions: https://medium.com/@jordan.l.edmunds/please-stop-hard-coding-file-paths-609c769f9537

**Comment by StanislavYatsevich** [comment]
> https://github.com/StanislavYatsevich/Store_Sales/blob/29e25693b941653d592edbd11e2a45ab4d626282/src/store_sales/streamlit_app.py#L7
> 
> Some of the possible solutions: https://medium.com/@jordan.l.edmunds/please-stop-hard-coding-file-paths-609c769f9537

https://github.com/StanislavYatsevich/Store_Sales/commit/3f99bd759c63845de2d01bd7d977b8ae739f9bee

**Comment by astraszab** [comment]
What will be pros and cons of having one constant per path and possible reusing the same constant in different scripts, instead of using four constants for two paths ("../../data/prepared_data" is duplicated in three constants: PREPARED_DATA_FOLDER_PATH, INPUT_DATA_FOLDER_PATH, and PROCESSED_DATA_FOLDER_PATH)? Which way do you think makes more sense for your usecase and why?

**Comment by StanislavYatsevich** [comment]
> What will be pros and cons of having one constant per path and possible reusing the same constant in different scripts, instead of using four constants for two paths ("../../data/prepared_data" is duplicated in three constants: PREPARED_DATA_FOLDER_PATH, INPUT_DATA_FOLDER_PATH, and PROCESSED_DATA_FOLDER_PATH)? Which way do you think makes more sense for your usecase and why?

As I found out, the choice of whether to use a single constant for all same-value paths or to use a constant for every path no matter what the value it has highly depends on the project and its requirements. For example, if it's needed to update the value of the path in the project, in the 1st case you will have to do it much less times than in the 2nd one. And if the paths are logically meant to be the same, using a single constant reflects that intention and ensures consistency across the project. 

On the other hand, if the logical structure of the project and paths diverges in future, it will be much easier to manage it with multiple independent constants. Another aspect is that if there are many different constants, their names will be more descriptive and informative, so you won't have to leave comments in your code or clarify it in README additionally.

I combined both of these approaches. I deleted one of the repeating constants, renamed them. So now my code uses one constant PREPARED_DATA_STAGE_1_FOLDER_PATH as the output path in the 1st stage and as the input path in the 2nd one since they are supposed to be the same. As for PREPARED_DATA_STAGE_2_FOLDER_PATH, this path isn't necessarily equal to the previous one, that's why I used another constant here. Clarified it in README.md

**Comment by astraszab** [comment]
Makes sense. Why did you chose these names for the constants (PREPARED_DATA_STAGE_1_FOLDER_PATH and PREPARED_DATA_STAGE_2_FOLDER_PATH)? What other options have you considered?

**Comment by StanislavYatsevich** [comment]
Actually because data preparation is performed in 2 stages, that's why I mentioned the stage number in the constants' names. And FOLDER_PATH in the end to emphasize that it's the path to the folder that's needed. I didn't come up with other better names for these constants.

**Comment by astraszab** [comment]
That's not practical for two reasons:
1. For a user, it's not clear what stage 1/2 means unless they go through the code and understand what each script does. The name should give the user some hint about the purpose of the folder
2. If later you decide to add one more stage before your current stage 1 or between stages 1 and 2, you'll need to rename the constant(s) in several code files to be consistent, and create a new constant with the same name, e.g., PREPARED_DATA_STAGE_1_FOLDER_PATH , but a different meaning because now stage 1 does something different.

Think of some names that wouldn't have these issues, based on what's actually stored in these folders

**Comment by StanislavYatsevich** [comment]
> That's not practical for two reasons:
> 
> 1. For a user, it's not clear what stage 1/2 means unless they go through the code and understand what each script does. The name should give the user some hint about the purpose of the folder
> 2. If later you decide to add one more stage before your current stage 1 or between stages 1 and 2, you'll need to rename the constant(s) in several code files to be consistent, and create a new constant with the same name, e.g., PREPARED_DATA_STAGE_1_FOLDER_PATH , but a different meaning because now stage 1 does something different.
> 
> Think of some names that wouldn't have these issues, based on what's actually stored in these folders

That's a good point. In the project prepared after the 1st stage data is used in the EDA and then some insights we get from it are used to prepare the data the final time, so I renamed these constants to PREPARED_FOR_EDA_DATA_FOLDER_PATH and PREPARED_FINAL_DATA_FOLDER_PATH respectively. I believe these names will be much more clear for a user.
c94004f
